plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftLaunchController.java

```diff
@@ -188,7 +192,9 @@ public void onDebugChange(DebuggingContext debuggingContext, IProgressMonitor mo
 					if( debuggerLaunch != null ) {
 						overrideHotcodeReplace(server, debuggerLaunch);
 					}
-				}					
+				} else if (isNodeJsProject(server)) {
```

Comment:
I think this check should be done in NodeDebugSessionTracker.
OpenShift launch controller just notifies all registered trackers and it's up to tracker to ignore the project or take any action.